### PR TITLE
Reword PR Template

### LINKED ISSUE #685: Rework The PR Template
## Description

The PR template has a lot going for it. I like the fields it has, and the descriptions written there for each one are quite adequate, yet, we still end up with a lot of PRs that either don't use the template and fail to provide information in any other way, or fail to fill in one or more fields that contain very important information. This applies to staff, core-devs, and non-staff contributors.

I won't list specific examples here as to not call anyone out, but I have seen:
- PRs with 2 or 3 word descriptions, that have popped out of nowhere, and are difficult to handle due to the missing context.
- PRs that fail to fill in the reason field, or fill it in incorrectly (see the gist linked below). I touched on it in [discord](https://discord.com/channels/267624335836053506/429409067623251969/824619053678067753), but that didn't go anywhere.
- PRs that are opened without issues, and cite the only reason behind the existence of the PR as being "discussed on discord" or something similar, with no links.

## Reasoning

Sir-lancebot is a project intended to be a gateway to open source contributions and is meant to make them as easy as possible. When someone opens a PR that meets one of the problems listed above, we are usually unable to review it because that missing context is important. We might end up with a lot of back and forth trying to extract those details, which is not fun for reviewers and contributors.

The intention of these changes is to create a standardized, easy-to-follow format, and enforce it in our guidelines.

## Proposed Implementation

I propose the following changes (do not make comments on the gist, keep them here):
[Gist](https://gist.githubusercontent.com/HassanAbouelela/0ca74e258caf57c07dd6f5e545e33803/raw/21ffe48383f59df84350848fd89289691d57f8a9/pull_request_template.md)

Update 1: Better explanation on when it is okay to make a PR without an issue. [Discussion](https://discord.com/channels/267624335836053506/411200599653351425/832591634952618004).
Update 2: Expand on what it means for an issue to be approved. [Discussion](https://discord.com/channels/267624335836053506/411200599653351425/832598212811554838).
Update 3: Rewrite the issues section to be shorter, and to the point. Discussion below.
Update 4: Final update - dropped the reasoning field.

Along with: 
- Adding a requirement to fill out the PR template to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/).
- Making the contributing guidelines easier to find on the site. Currently, the sir-lancebot getting started guides has a link to the contributing file on github, which just redirects to the contributing guide on the site.

## Would you like to implement this yourself?

Not really relevant, this will be more of a discussion anyways.

### PR DESCRIPTION
**Pull request by HassanAbouelela**
This is a rewording of the comments in the PR template, to hopefully clarify intentions.
It also removes unneeded fields.

## Relevant Issues

Closes #685.

## Description

Rewords portions of the PR template to clarify intentions through comments, and removing unneeded fields.

Make sure to view the raw file to see the comments.

## Reasoning

This field is gone, shoo.

## Screenshots

This field is gone as well.

## Additional Details

This one too.

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] If dependencies have been added or updated, run `pipenv lock`?
- [x] **Lint your code** (`pipenv run lint`)?
- [x] Set the PR to **allow edits from contributors**?

### CODE REVIEW

File: .github/pull_request_template.md
Comment (Akarys42) [inline .github/pull_request_template.md:3]: I had to read this sentence twice to understand it, does this read better? 
```suggestion
Linking an issue approved by a Core Developer, that has the "approved" label, is mandatory.
```
Maybe the commas aren't necessary too

**Review by Akarys42 — APPROVED** [review]
Lgtm, thanks scal! I just have a comment on one of the lines

File: .github/pull_request_template.md
Comment (scragly) [reply .github/pull_request_template.md:3]: Tbh the wording sounds a bit unnatural in general.

> It is mandatory to link an issue with the "approved" label. 

We don't really need to outline that it's approved by a Core Dev, when the label already says that in it's description, do we?

Comment (HassanAbouelela) [reply .github/pull_request_template.md:3]: The reason for adding it in the first place was to address concerns about who is approving issues. Currently, anyone with write permissions (all staff members) can add the approved label. The conversation was linked in the issue, but [here](https://discord.com/channels/267624335836053506/411200599653351425/832591634952618004) it is again (staff channel). I don't mind it either way, but the consensus at the time was approving issues is a core dev responsibility, we should outline that.

Comment (scragly) [reply .github/pull_request_template.md:3]: Okay. In that case, it's really only getting rid of the inner phrasing to make it flow better while keeping the same info.

> It is mandatory to link to an issue that has been approved by a Core Developer, indicated by an "approved" label.

Comment (HassanAbouelela) [reply .github/pull_request_template.md]: Addressed in 525078b

File: .github/pull_request_template.md
Comment (Xithrius) [inline .github/pull_request_template.md:18]: How will we be encouraging linting, if at all?

**Review by Xithrius — COMMENTED** [review]
Looks good, besides one singular question.

File: .github/pull_request_template.md
Comment (Akarys42) [reply .github/pull_request_template.md:18]: It is mentioned inside the contrib guidelines. It may be worth to mention them here again but in the end the failed lint will be the same thing, we’ll end up with a second fix commit.

Comment (Xithrius) [reply .github/pull_request_template.md:18]: Maybe we should link the contributing guidelines in a comment within this PR template?

Comment (Akarys42) [reply .github/pull_request_template.md:18]: It is already mentioned line 20 but I feel like it could even be the first line of the template. We could also add a sentence about when this template should be used.

Comment (Xithrius) [reply .github/pull_request_template.md:18]: I somehow completely missed that link with my eyes. Sorry about that, this PR looks good to me.

**Review by Xithrius — APPROVED** [review]

**Pull request merged**
